pkg/storage/fs/ocis/persistence.go

```diff
@@ -0,0 +1,36 @@
+package ocis
+
+import (
+	"context"
+	"net/url"
+	"os"
+
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+)
+
+// TreePersistence is used to manage a tree hierarchy
+type TreePersistence interface {
+	GetPathByID(ctx context.Context, id *provider.ResourceId) (string, error)
+	GetMD(ctx context.Context, node *NodeInfo) (os.FileInfo, error)
+	ListFolder(ctx context.Context, node *NodeInfo) ([]*NodeInfo, error)
+	CreateDir(ctx context.Context, node *NodeInfo) (err error)
+	CreateReference(ctx context.Context, path string, targetURI *url.URL) error
+	Move(ctx context.Context, oldNode *NodeInfo, newNode *NodeInfo) (err error)
+	Delete(ctx context.Context, node *NodeInfo) (err error)
+
+	Propagate(ctx context.Context, node *NodeInfo) (err error)
+}
+
+// PathWrapper is used to encapsulate path transformations
+type PathWrapper interface {
```

Comment:
streamline this nicer separation of interfaces for all storages ? (could be a separate task/ticket)

Reply:
future work, api still needs some refactoring